.github/workflows/new_release.yml

```
@@ -13,26 +13,47 @@ on:
         description: "Run all tests (true/false)"
         required: true
 jobs:
-    release:
-      runs-on: ubuntu-20.04
-      outputs:
-        url: ${{ steps.create_release.outputs.upload_url }}
-      steps:
-        - name: Release creation
-          uses: actions/create-release@v1
-          id: create_release
-          env:
-            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          with:
-            tag_name: ${{ github.event.inputs.release_tag }}
-            release_name: ${{ github.event.inputs.release_name }}
-            body: |
-              Please complete description
+      release:
+        name: Release pushed tag
+        runs-on: ubuntu-20.04
+        steps:
+          - name: Create release
+            env:
+              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+              tag: ${{ github.event.inputs.release_tag }}
+              title: ${{ github.event.inputs.release_name }}
+            run: |
+              gh release create "$tag" \
+                  --repo="$GITHUB_REPOSITORY" \
+                  --title="$title" \
+                  --generate-notes        
```

```suggestion
            env:
              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
              tag: ${{ github.event.inputs.release_tag }}
              title: ${{ github.event.inputs.release_name }}
              tests: ${{ github.event.inputs.run-tests }}
            run: |
              gh release create "$tag" \
                  --repo="$GITHUB_REPOSITORY" \
                  --title="$title" \
                  --notes="Run tests: $tests"
```

The generated notes are garbage since our history is not linear. Also, we want to sort changes by type (new feature, bugfix, doc, code cleaning, etc.)